yang/frr-bgp-filter.yang

```diff
@@ -0,0 +1,341 @@
+module frr-bgp-filter {
+  yang-version 1.1;
+  namespace "http://frrouting.org/yang/bgp-filter";
+  prefix frr-bgp-filter;
+
+  import frr-filter {
+    prefix frr-filter;
+  }
+
+  import ietf-routing-types {
+    prefix rt-types;
+  }
+
+  organization
+    "Free Range Routing";
+  contact
+    "FRR Users List:       <mailto:frog@lists.frrouting.org>
+     FRR Development List: <mailto:dev@lists.frrouting.org>";
+  description
+    "This module defines filter settings";
+
+  revision 2020-01-15 {
+    description
+      "Initial revision";
+  }
+
+  typedef list-sequence {
+    type uint32 {
+      range "1..4294967295";
+    }
+    description
+      "List instance priority (low number means higher priority)";
+  }
+
+  typedef list-action {
+    type enumeration {
+      enum "deny" {
+        value 0;
+        description
+          "Deny an entry";
+      }
+      enum "permit" {
+        value 1;
+        description
+          "Accept an entry";
+      }
+    }
+    description
+      "Return action on match";
+  }
+
+  typedef bgp-list-standard {
+    type uint8 {
+      range "1..99";
+    }
+    description
+      "Standard list";
+  }
+
+  typedef bgp-list-extended {
+    type uint16 {
+      range "100..500";
+    }
+    description
+      "Extended list";
+  }
+
+  typedef bgp-list-name {
+    type string;
+    description
+      "List name";
+  }
+
+  typedef community-string {
+    type string {
+      pattern "(6553[0-5]|655[0-2][0-9]|65[0-4][0-9]{2}|6[0-4][0-9]{3}|[1-5][0-9]{4}|[1-9][0-9]{0,3}|0)|((6553[0-5]|655[0-2][0-9]|65[0-4][0-9]{2}|6[0-4][0-9]{3}|[1-5][0-9]{4}|[1-9][0-9]{0,3}|0):(6553[0-5]|655[0-2][0-9]|65[0-4][0-9]{2}|6[0-4][0-9]{3}|[1-5][0-9]{4}|[1-9][0-9]{0,3}|0))|(local-AS)|(no-advertise)|(no-export)|(internet)";
```

Comment:
Should `additive` be one of the options here?

Also, why is `AS` capitalized in `local-AS`?